### PR TITLE
Refuse to load overly large manifests

### PR DESCRIPTION
**Pull request by swankjesse**

### CODE REVIEW

File: zipline-loader/src/commonMain/kotlin/app/cash/zipline/loader/internal/fetcher/LoadedManifest.kt [hunk 1]
```diff
@@ -36,25 +38,48 @@ data class LoadedManifest(
...
 
   // Because new releases may change default values, it's best to encode them.
   encodeDefaults = true
 }
 
+/**
+ * Confirm the manifest is of a reasonable size before proceeding to operate on it. 10 KiB is a
+ * typical size for our test applications. 640K ought to be enough for anybody.
+ */
+const val MANIFEST_MAX_SIZE = 640 * 1024
```
Comment (JakeWharton) [inline zipline-loader/src/commonMain/kotlin/app/cash/zipline/loader/internal/fetcher/LoadedManifest.kt:58]: Meant to be public?

**Review by JakeWharton — APPROVED** [review]
Hard to review on mobile

File: zipline-loader/src/commonMain/kotlin/app/cash/zipline/loader/internal/fetcher/LoadedManifest.kt (hunk 1 above)
Comment (swankjesse) [reply zipline-loader/src/commonMain/kotlin/app/cash/zipline/loader/internal/fetcher/LoadedManifest.kt:58]: Yeah, we’re using it from the CLI module. I still don’t really have good ideas for internal APIs that can’t be internal visibility

Comment (JakeWharton) [reply zipline-loader/src/commonMain/kotlin/app/cash/zipline/loader/internal/fetcher/LoadedManifest.kt:58]: There are friend paths which make `internal` work across compilations, but it's complicated to set up. It's how test sources see main source's `internal` things today.

This is the feature request for the good UX version: https://youtrack.jetbrains.com/issue/KT-22284/

File: zipline-loader/src/commonMain/kotlin/app/cash/zipline/loader/internal/fetcher/HttpFetcher.kt [hunk 2]
```diff
@@ -50,19 +50,22 @@ internal class HttpFetcher(
     url: String,
     freshAtEpochMs: Long,
   ): LoadedManifest {
-    val manifestBytesWithoutBaseUrl = fetchByteString(applicationName, null, url)
+    val manifestBytesWithoutBaseUrlUtf8 = fetchByteString(applicationName, null, url)
```
Comment (nmahendru) [inline zipline-loader/src/commonMain/kotlin/app/cash/zipline/loader/internal/fetcher/HttpFetcher.kt:53]: How about we just limit the download size here somehow ? We should not download it in the first place if it's > MANIFEST_MAX_SIZE.

File: zipline-loader/src/commonTest/kotlin/app/cash/zipline/loader/LoaderTester.kt [hunk 3]
```diff
@@ -148,6 +150,32 @@ class LoaderTester(
     )
   }
 
+  suspend fun failureManifestTooLarge(applicationName: String): String {
```
Comment (nmahendru) [inline zipline-loader/src/commonTest/kotlin/app/cash/zipline/loader/LoaderTester.kt:153]: 👏 great tests.

**Pull request merged**